src/Database/Expression/CaseStatementExpression.php

```diff
@@ -102,48 +101,52 @@ class CaseStatementExpression implements CaseExpressionInterface
     /**
      * Constructor.
      *
-     * @param \Cake\Database\TypeMap|null $typeMap The type map to use when using an array of conditions for the `WHEN`
-     *  value.
+     * When a value is set, the syntax generated is
+     * `CASE case_value WHEN when_value ... END` (simple case),
+     * where the `when_value`'s are compared against the
+     * `case_value`.
+     *
+     * When no value is set, the syntax generated is
+     * `CASE WHEN when_conditions ... END` (searched case),
+     * where the conditions hold the comparisons.
+     *
+     * Note that `null` is a valid case value, and thus should
+     * only be passed if you actually want to create the simple
+     * case expression variant!
+     *
+     * @param \Cake\Database\ExpressionInterface|object|scalar|null $value The case value.
+     * @param string|null $valueType The case value type. If no type is provided, the type will be tried to be inferred
+     *  from the value.
      */
-    public function __construct(?TypeMap $typeMap = null)
+    public function __construct($value = null, ?string $valueType = null)
```

Comment:
```suggestion
    public function __construct($value = null, ?string $type= null)
```
This can probably just be `type` unless you need it named like this.

Reply:
It certainly can be.